api/app/controllers/api/v1/admin/listing_complexes_controller.rb

```
@@ -0,0 +1,205 @@
+            return
+          end
+
+          url = params[:listing_complex]&.[](:url)
+
+          if url.blank?
+            render json: { errors: ['URL é obrigatória'] }, status: :unprocessable_entity
+            return
+          end
+
+          # Validate URL starts with tenant's scraper source
+          scraper_domain = URI.parse(tenant.scraper_source_url).host rescue nil
```

RuboCop (1.81.1)

[convention] Avoid using `rescue` in its modifier form.

(Style/RescueModifier)